pkg/rclone/copy.go

```diff
@@ -0,0 +1,445 @@
+// Copyright 2025 Clyso GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rclone
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	awss3 "github.com/aws/aws-sdk-go/service/s3"
```

Comment:
since it is now also depends on s3 sdk, should we rename package from rclone to something more generic?

Reply:
I've put it here, so the logic related to the object move would be more or less in the one place. the moment we'll replace rclone also in regular migration task, we'd be able to drop it all together.